File: app/src/main/java/com/maple/audiometry/ui/activity/MainActivity.java
package com.maple.audiometry.ui.activity;

import android.Manifest;
import android.content.Intent;
import android.os.Bundle;
import android.view.KeyEvent;
import android.view.View;

import com.maple.audiometry.R;
import com.maple.audiometry.base.BaseFragmentActivity;
import com.maple.audiometry.utils.permission.PermissionFragment;
import com.maple.msdialog.AlertDialog;

import butterknife.ButterKnife;
import butterknife.OnClick;

/**
 * 功能主界面
 *
 * @author shaoshuai
 */
public class MainActivity extends BaseFragmentActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        ButterKnife.bind(this);


        PermissionFragment.getPermissionFragment(this)
                .checkPermissions(new String[]{
                        Manifest.permission.WRITE_EXTERNAL_STORAGE,
                        Manifest.permission.RECORD_AUDIO
                }, null);
    }

    // 去检测噪音
    @OnClick(R.id.noise)
    public void toCheckNoise() {
        Intent intent = new Intent(this, NoiseCheckActivity.class);
        startActivity(intent);
    }

    // 去检查耳朵
    @OnClick(R.id.voice)
    public void toCheckEar() {
        Intent intent = new Intent(this, VoiceActivity.class);
        startActivity(intent);
    }
    // 1024398
    @Override
    public boolean onKeyDown(int keyCode, KeyEvent event) {
        if (keyCode == KeyEvent.KEYCODE_BACK && event.getRepeatCount() == 0) {
            new AlertDialog(this)
                    .setScaleWidth(0.7)
                    .setTitle("是否退出？")
                    .setLeftButton("取消", null)
                    .setRightButton("退出", new View.OnClickListener() {
                        @Override
                        public void onClick(View v) {
                            finish();
                        }
                    })
                    .show();
        }
        return false;
    }

}

File: app/src/main/java/com/maple/audiometry/ui/home/MainActivity.kt
package com.maple.audiometry.ui.home

import android.content.Intent
import android.os.Bundle
import com.maple.audiometry.R
import com.maple.audiometry.ui.base.BaseFragmentActivity
import com.maple.audiometry.ui.detection.DetectionActivity
import com.maple.audiometry.ui.noise.NoiseCheckActivity
import com.maple.audiometry.utils.T
import kotlinx.android.synthetic.main.activity_main.*

/**
 * 功能主界面
 *
 * @author maple
 */
class MainActivity : BaseFragmentActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        bt_noise.setOnClickListener { toCheckNoise() }
        bt_voice.setOnClickListener { toCheckEar() }

    }

    // 去检测噪音
    private fun toCheckNoise() {
        val intent = Intent(this, NoiseCheckActivity::class.java)
        startActivity(intent)
    }

    // 去检查耳朵
    private fun toCheckEar() {
        val intent = Intent(this, DetectionActivity::class.java)
        startActivity(intent)
    }

    private var exitTime: Long = 0
    override fun onBackPressed() {
        if (System.currentTimeMillis() - exitTime > 2000) {
            T.showShort(mContext, "再按一次退出程序")
            exitTime = System.currentTimeMillis()
        } else {
            finish()
        }
    }

}
